engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -240,16 +240,10 @@ private ConstrainedField findPropertyMetaData(Field field) {
 		);
 
 		boolean isCascading = field.isAnnotationPresent( Valid.class );
-		UnwrapValidatedValue unwrapValidatedValue = field.getAnnotation( UnwrapValidatedValue.class );
-		UnwrapMode unwrapMode = UnwrapMode.AUTOMATIC;
-		if ( unwrapValidatedValue != null ) {
-			unwrapMode = unwrapValidatedValue.value() ? UnwrapMode.UNWRAP : UnwrapMode.SKIP_UNWRAP;
-		}
-
 		Set<MetaConstraint<?>> typeArgumentsConstraints = findTypeAnnotationConstraintsForMember( field );
-		if ( !typeArgumentsConstraints.isEmpty() && !ReflectionHelper.isIterable( ReflectionHelper.typeOf( field ) ) ) {
-			unwrapMode = UnwrapMode.UNWRAP;
-		}
+
+		boolean typeArgumentAnnotated = !typeArgumentsConstraints.isEmpty();
+		UnwrapMode unwrapMode = unwrapMode( field, typeArgumentAnnotated );
```

Comment:
`UnwrapMode` tells your whether or not the field should be unwrapped (basically the value of `@UnwrapValidatedValue`). The change will enforce unwrapping in case there are type use constraints. Note, that we basically "roll" type use constraints into the field metadata. The exact unwrapping behavior will be determined at validation time.